mesh/mesh.go

```diff
@@ -754,11 +768,25 @@ func (msh *Mesh) accumulateRewards(l *types.Layer, params Config) {
 			msh.With().Warning("Atx from block not found in db", log.Err(err), bl.ID(), bl.ATXID)
 			continue
 		}
-		ids = append(ids, atx.Coinbase)
-
+		coinbases = append(coinbases, atx.Coinbase)
+		//create a map where we count the number of coinbase-smesherID pairs appearing
+		if _, exists := coinbasesAndSmeshers[atx.Coinbase]; !exists {
+			coinbasesAndSmeshers[atx.Coinbase] = make(map[string]uint64)
+			coinbasesAndSmeshers[atx.Coinbase][atx.NodeID.String()]++
+		} else {
+			coinbasesAndSmeshers[atx.Coinbase][atx.NodeID.String()]++
+		}
+		if _, exists := smeshers[atx.Coinbase]; !exists {
+			smeshers[atx.Coinbase] = make([]types.NodeID, 0)
+			smeshers[atx.Coinbase] = append(smeshers[atx.Coinbase], atx.NodeID)
+		} else {
+			if !checkIfPresent(atx.NodeID, smeshers[atx.Coinbase]) {
+				smeshers[atx.Coinbase] = append(smeshers[atx.Coinbase], atx.NodeID)
+			}
+		}
 	}
 
-	if len(ids) == 0 {
+	if len(coinbasesAndSmeshers) == 0 {
```

Comment:
Why does this use `coinbasesAndSmeshers` but a few lines below you get `numBlocks` from `coinbases`?